storage/storage.go

```diff
@@ -88,9 +103,19 @@ func New(config Config) (Service, error) {
 				return nil, microerror.MaskAnyf(invalidConfigError, "etcd address must not be empty")
 			}
 
+			transport := &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+				Dial: (&net.Dialer{
+					Timeout:   30 * time.Second,
+					KeepAlive: 30 * time.Second,
+				}).Dial,
+				TLSHandshakeTimeout: 10 * time.Second,
+				TLSClientConfig:     tlsConfig,
+			}
```

Comment:
This is the same as `client.DefaultTransport`. But `client.DefaultTransport` hides it behind `CancellableTransport` interface so `TLSClientConfig` is inaccessible.

I could go with type assertion, but I like this approach more because it fails at compile time instead of runtime.